scripts/deploy-image.sh

```diff
@@ -0,0 +1,40 @@
+#!/usr/bin/env bash
+
+set -o errexit   # abort on nonzero exitstatus
+set -o nounset   # abort on unbound variable
+set -o pipefail  # don't hide errors within pipes
+
+required_env_vars=(
+  CI_REGISTRY_IMAGE
+)
+
+for var in "${required_env_vars[@]}"; do
+  if [ -z ${!var+x} ]; then
```

Comment:
Here we directly check if the variable is set, and not just whether it is also empty. If set to empty string, that's still a legitimate setting. (It could be wrong in a different way though).

https://stackoverflow.com/questions/3601515/how-to-check-if-a-variable-is-set-in-bash